pkg/memtier/counters_heatmap.go

```diff
@@ -235,7 +242,7 @@ func (h *Heatmap) updateFromPidHeatRange(pid int, thr *HeatRange) {
 				created: hr.created,
 				updated: hr.updated,
 			}
-			*hrs = append(*hrs, &newHr)
+			(*hrs).Set(newHr.addr, &newHr)
 			hr.addr = thr.addr
```

Comment:
When overlapping ranges are modified (`hr.addr = ....`), it changes the key (the address) of an existing address range in hrs. That is, the rbtree needs to be updated accordingly.